module-system/module-implementations/sov-chain-state/src/tests.rs

```diff
@@ -0,0 +1,185 @@
+use sov_data_generators::value_setter_data::ValueSetterMessages;
+use sov_data_generators::{has_tx_events, new_test_blob_from_batch, MessageGenerator};
+use sov_modules_api::default_context::DefaultContext;
+use sov_modules_api::Spec;
+use sov_modules_stf_template::{AppTemplate, SequencerOutcome};
+use sov_rollup_interface::mocks::{
+    MockZkvm, TestBlob, TestBlock, TestBlockHeader, TestHash, TestValidityCond,
+};
+use sov_rollup_interface::stf::StateTransitionFunction;
+use sov_state::{ProverStorage, Storage};
+
+use crate::tests_helpers::{create_demo_genesis_config, get_working_set, TestRuntime};
+use crate::{StateTransitionId, TransitionInProgress};
+
+type C = DefaultContext;
+
+/// This test generates a new mock rollup having a simple value setter module
+/// with an associated chain state, and checks that the height, the genesis hash
+/// and the state transitions are correctly stored and updated.
+#[test]
+fn test_simple_value_setter() {
+    // Build an app template with the module configurations
+    let runtime = TestRuntime::default();
+
+    let tmpdir = tempfile::tempdir().unwrap();
+
+    let storage: ProverStorage<sov_state::DefaultStorageSpec> =
+        ProverStorage::with_path(tmpdir.path()).unwrap();
+
+    let mut app_template = AppTemplate::<
```

Comment:
This is my personal opinion, 
but it might be simpler to tests hooks directly, without initiating whole app template, creating proper runtime for it, etc.

Reply:
I think we want to aim for that kind of testing framework for the modules after discussion with @bkolad.

Reply:
My understanding was that `AppTemplate` was needed because we had to run the whole `apply_slot`. I am fine with jumping on a zoom call to discuss it further (but maybe whatever we decide should go as another issue/PR so it won't block this big PR)